ibm/flex/structures.go

```diff
@@ -892,6 +893,61 @@ func ReplicationRuleGet(in *s3.ReplicationConfiguration) []map[string]interface{
 	return rules
 }
 
+func ObjectLockConfigurationGet(in *s3.ObjectLockConfiguration) []map[string]interface{} {
+	fmt.Println("*** ObjectLockConfigurationGet from structure***")
+	configuration := make([]map[string]interface{}, 0, 1)
```

Comment:
Here we are defining the capacity as 1 because we already know that the max item in this array are going to be 1 (as only 1 object lock configuration can be put on a bucket).This way the memory size allocated to it remains constant i.e. 1. If we make it zero and happen if not nil , as the capacity is not defined the make call sometimes does unnecessary memory allocation which can be avoided by providing the capacity.

Reply:
We can do it the way you said , but this way we avoid the unnecessary memory allocation and wasting any extra memory.